flink-python/pyflink/datastream/connectors/file_system.py

```diff
@@ -526,8 +539,15 @@ class FileSink(Sink):
     the checkpoint from which we restore.
     """
 
-    def __init__(self, j_file_sink):
+    def __init__(self, j_file_sink, preprocessing=None):
         super(FileSink, self).__init__(sink=j_file_sink)
+        self._preprocessing = preprocessing
+
+    def need_preprocessing(self):
+        return self._preprocessing is not None
+
+    def get_preprocessing(self):
```

Comment:
What about refactoring SupportPreprocessing as following:
```
def apply(ds):
  pass
```

Then we could simply the implementation and also avoid introducing TransformAppender.